winpr/libwinpr/sspi/NTLM/ntlm.c

```diff
@@ -313,10 +314,30 @@ static SECURITY_STATUS SEC_ENTRY ntlm_AcquireCredentialsHandleW(
 	credentials->fCredentialUse = fCredentialUse;
 	credentials->pGetKeyFn = pGetKeyFn;
 	credentials->pvGetKeyArgument = pvGetKeyArgument;
+
 	identity = (SEC_WINNT_AUTH_IDENTITY*)pAuthData;
 
 	if (identity)
+	{
 		sspi_CopyAuthIdentity(&(credentials->identity), identity);
+		if (identity->Flags & SEC_WINNT_AUTH_IDENTITY_EXTENDED)
+			settings = &((SEC_WINNT_AUTH_IDENTITY_WINPR*)pAuthData)->ntlmSettings;
+	}
```

Comment:
As we're in the unicode variant of the function another possibility (than adding a new flag) would be that `pAuthData` would be a kind of `SEC_WINNT_AUTH_IDENTITY_EXW` (a NTLM specialized one) where you can distinguish the various kind by looking at the `Version` and `Length` fields (that's what they do in win32 implementations). Anyway that has the drawback that the ascii version of the function have to build a unicode version of the parameters.
Your design is probably the best, I just wanted to share my thoughts.

Reply:
True. Since there's a flags field anyway we may as well use it... makes for cleaner code IMHO